src/goto-symex/symex_builtin_functions.cpp

```diff
@@ -167,6 +169,27 @@ void goto_symext::symex_malloc(
 
   new_symbol_table.add(value_symbol);
 
+  exprt zero_init=code.op1();
+  state.rename(zero_init, ns); // to allow constant propagation
+  simplify(zero_init, ns);
+
+  if(zero_init.is_constant() && !zero_init.is_zero())
```

Comment:
Shouldn't we throw an exception if `zero_init.is_constant()` does not hold instead of silently ignoring the violated precondition?